pandas/compat/_optional.py

```diff
@@ -72,6 +72,119 @@
     "tables": "pytables",
 }
 
+# Mapping of operation contexts to alternative dependencies
+OPERATION_CONTEXTS = {
+    "excel": {
+        "alternatives": ["openpyxl", "xlsxwriter", "calamine", "xlrd", "pyxlsb", "odfpy"],
```

Comment:
These may be alternatives, but they may not be. They are not completely equivalent. If a user is doing `.write_excel(..., engine="openpyxl")`, I don't think we should be suggesting they try installing `xlsxwriter`.